src/main/java/net/snowflake/client/core/bind/BindUploader.java

```diff
@@ -159,16 +159,18 @@ private synchronized String synchronizedTimestampFormat(String o, String type) {
     int nano = times.right;
 
     Timestamp v1 = new Timestamp(sec * 1000);
+    ZoneOffset offsetId;
     // For timestamp_ntz, use UTC timezone. For timestamp_ltz, use the local timezone to minimise
     // the gap.
     if ("TIMESTAMP_LTZ".equals(type)) {
       TimeZone tz = TimeZone.getDefault();
       cal.setTimeZone(tz);
       cal.clear();
       timestampFormat.setCalendar(cal);
+    } else {
+      offsetId = ZoneOffset.UTC;
     }
-
-    ZoneOffset offsetId = ZoneId.systemDefault().getRules().getOffset(Instant.now());
+    offsetId = ZoneId.systemDefault().getRules().getOffset(Instant.ofEpochMilli(v1.getTime()));
```

Comment:
we override here the offset set in else branch - it does not look corect

Reply:
Fixed.